OutOfSchool/OutOfSchool.AuthCommon/Controllers/AuthController.cs

```diff
@@ -397,13 +395,13 @@ public async Task<IActionResult> Register(RegisterViewModel model)
                 {
                     await signInManager.SignInAsync(user, false);
 
-                    if (user.Role == Role.Parent.ToString().ToLower())
+                    if (user.Role.Equals(nameof(Role.Parent), StringComparison.OrdinalIgnoreCase))
                     {
                         var parent = new Parent()
                         {
                             UserId = user.Id,
-                            Gender = model.Gender,
-                            DateOfBirth = model.DateOfBirth,
+                            Gender = Gender.Male,
+                            DateOfBirth = DateTime.Now.AddYears(-18),
```

Comment:
> (Code clarity) This is bit unclear why all parents are 18 yrs old males, until you read the PR description. Later, in the plain code, there will be no PR description, thus it would be even more obscure. Please consider a way to shed more light here.
> 
> In example:
> 
> * either put a comment that you are creating some default parent with specific id and other valid fileds
> * or create helper method in `Parent` class with appropriate name like `CreateDefaultWithUserId(Guid userId)`
> 
> As for me latter is better, because of encapsulation (better reuse, refactoring, testing, search), and because I don't like comments in the code (there are a few reasons)

I completely agree with you that creating all parents as 18-year-old males is unclear. I’ve considered a few solutions:

1. **Enum Modification:** Initially, I thought about adding a `None` case to the `Gender` enum. This would make things clearer, but it would require changing the entire existing parent structure in the database. Currently, we have `Male` as the default enum value, so introducing `None` would require adjustments. However, this approach wouldn’t fix the issue of parent age.

2. **Remove Required Attribute:** Another option is to remove the `Required` attribute from the properties within the `Parent` class. While this approach is less harmful to the database than the first one, we need to be careful when working with these properties in the future.

3. **Plug Values (current int this PR):** The simplest approach would be to set these properties to some default values (such as the current ones). However, as you rightly pointed out, this approach lacks clarity.

Given these options, should we create a factory method for the default parent inside the `Parent` class, or try implement other variants?

Reply:
Mentioned options:
1. `None` case in `enum` - has it's own downsides, like `null` for reference type, there is no compile-time check for valid/existing data, thus you either checks in code or ignore them (or forget) and have run-time issues.
2. No `Required` attribute and nullable properties - breaks encapsulation, same absence of compile-time check if `Parent` is valid, need additional `if`s to enforce this, same issues as for previous
3. Dummy values - encapsulation is better, but data are still "wrong", you can't trust age/gender data taken from any `Parent`. Compiler is happy, but you should not forget to put real data for specific `Parent` later in the code.

So either option is "bad". 

Solution could be introducing of separate `DraftParent` class (without gender and age) in addition to existing `Parent` class (with mentioned properties). Benefit - compile-time control of correctness, clear point in the code where data are converted from one type to another. Downside (generic for many "design-changing" solution) - wider changes of code, more entities to support.

That's why here I suggest to take "bad" solution (current), but at least document it correctly, using "factory" method from `Parent`. 

Ofc, you could check that additional type introducing approach, maybe it won't introduce many changes.